java/adapter/jdbc/src/test/java/org/apache/arrow/adapter/jdbc/JdbcToArrowTestHelper.java

```diff
@@ -56,112 +56,167 @@ public static void assertIntVectorValues(IntVector intVector, int rowCount, Inte
+      if (values[j] == null) {
+        assertTrue(float8Vector.isNull(j));
+      } else {
+        assertEquals(values[j], float8Vector.get(j), 0.01);
+      }
     }
   }
 
   public static void assertFloat4VectorValues(Float4Vector float4Vector, int rowCount, Float[] values) {
     assertEquals(rowCount, float4Vector.getValueCount());
 
     for (int j = 0; j < float4Vector.getValueCount(); j++) {
-      assertEquals(values[j], float4Vector.get(j), 0.01);
+      if (values[j] == null) {
+        assertTrue(float4Vector.isNull(j));
+      } else {
+        assertEquals(values[j], float4Vector.get(j), 0.01);
+      }
     }
   }
 
   public static void assertTimeVectorValues(TimeMilliVector timeMilliVector, int rowCount, Long[] values) {
     assertEquals(rowCount, timeMilliVector.getValueCount());
 
     for (int j = 0; j < timeMilliVector.getValueCount(); j++) {
-      assertEquals(values[j].longValue(), timeMilliVector.get(j));
+      if (values[j] == null) {
+        assertTrue(timeMilliVector.isNull(j));
+      } else {
+        assertEquals(values[j].longValue(), timeMilliVector.get(j));
+      }
     }
   }
 
   public static void assertDateVectorValues(DateMilliVector dateMilliVector, int rowCount, Long[] values) {
     assertEquals(rowCount, dateMilliVector.getValueCount());
 
     for (int j = 0; j < dateMilliVector.getValueCount(); j++) {
-      assertEquals(values[j].longValue(), dateMilliVector.get(j));
+      if (values[j] == null) {
+        assertTrue(dateMilliVector.isNull(j));
+      } else {
+        assertEquals(values[j].longValue(), dateMilliVector.get(j));
+      }
     }
   }
 
   public static void assertTimeStampVectorValues(TimeStampVector timeStampVector, int rowCount, Long[] values) {
     assertEquals(rowCount, timeStampVector.getValueCount());
 
     for (int j = 0; j < timeStampVector.getValueCount(); j++) {
-      assertEquals(values[j].longValue(), timeStampVector.get(j));
+      if (values[j] == null) {
+        assertTrue(timeStampVector.isNull(j));
+      } else {
+        assertEquals(values[j].longValue(), timeStampVector.get(j));
+      }
     }
   }
 
   public static void assertVarBinaryVectorValues(VarBinaryVector varBinaryVector, int rowCount, byte[][] values) {
     assertEquals(rowCount, varBinaryVector.getValueCount());
 
     for (int j = 0; j < varBinaryVector.getValueCount(); j++) {
-      assertArrayEquals(values[j], varBinaryVector.get(j));
+      if (values[j] == null) {
+        assertTrue(varBinaryVector.isNull(j));
+      } else {
+        assertArrayEquals(values[j], varBinaryVector.get(j));
+      }
     }
   }
 
   public static void assertVarcharVectorValues(VarCharVector varCharVector, int rowCount, byte[][] values) {
     assertEquals(rowCount, varCharVector.getValueCount());
 
     for (int j = 0; j < varCharVector.getValueCount(); j++) {
-      assertArrayEquals(values[j], varCharVector.get(j));
+      if (values[j] == null) {
```

Comment:
it is possible to factor out the the code in some way to avoid the same if/else block in all tests?

Reply:
Since the forms(values/vector.get(j)/assert) are not all same, I think it's hard to factor out or make it complex to read.